File: javascript/leetcode/binary-tree-tilt.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var findTilt = function(root) {
    if (root === null) { return 0; }
    var tilt = Math.abs(findSum(root.left) - findSum(root.right));
    return tilt + findTilt(root.left) + findTilt(root.right);
};

var findSum = function(root) {
    if (root === null) { return 0; }
    return root.val + findSum(root.left) + findSum(root.right);
}

File: java/leetcode/binary-tree-tilt.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public int findTilt(TreeNode root) {
        if (root == null) { return 0; }
        int tilt = Math.abs(findSum(root.left) - findSum(root.right));
        return tilt + findTilt(root.left) + findTilt(root.right);
    }

    public int findSum(TreeNode root) {
        if (root == null) { return 0; }
        return root.val + findSum(root.left) + findSum(root.right);
    }
}
